frontend/main.js

```diff
@@ -206,6 +207,9 @@ async function startApp () {
     return
   }
 
+  // register service-worker
+  await sbp('service-workers/setup')
```

Comment:
I placed this sbp call here only because it's before the app Vue instance is created but let me know if you think it has to be moved elsewhere.

Reply:
Probably OK? 🤷 

However this reminds me about this comment you left:

> While reviewing, make sure to unregister the service-worker if there is already one installed from reviewing previously so that it is re-installed again.

Isn't there a way to automatically upgrade the service worker?

Reply:
@taoeffect 
If you look at [this article](https://web.dev/articles/service-worker-lifecycle#updates) written about service-worker's lifecycle, the service-worker update is detected and executed automatically by the browser when the user visits the website. 

Also, there seems to be a method that we can use to manually do that when we want, which is [here](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/update).

(Sorry for a late reply, needed to take some time for researching on this topic.)

Reply:
It looks like there are a few APIs we might want to consider:

> You can take control of uncontrolled clients by calling `clients.claim()` within your service worker once it's activated.

> The waiting phase means you're only running one version of your site at once, but if you don't need that feature, you can make your new service worker activate sooner by calling `self.skipWaiting()`.

> If you expect the user to be using your site for a long time without reloading, you may want to call `update()` on an interval (such as hourly).

Perhaps for a subsequent PR

Reply:
Created and assigned a new issue for that: https://github.com/okTurtles/group-income/issues/1791